src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs

```diff
@@ -24,17 +24,7 @@ namespace Windows.UI.Xaml.Controls.Primitives;
 
 internal partial class PopupPanel : Panel
 {
-	private ManagedWeakReference _popup;
-
-	public Popup Popup
-	{
-		get => _popup?.Target as Popup;
-		set
-		{
-			WeakReferencePool.ReturnWeakReference(this, _popup);
-			_popup = WeakReferencePool.RentWeakReference(this, value);
-		}
-	}
+	public Popup Popup { get; }
```

Comment:
@Youssef1313 This is likely going to cause memory leaks on iOS/Android. Why do you think this is needed? If this is for a performance issue, we can use the already existing compiler constant `UNO_HAS_UIELEMENT_IMPLICIT_PINNING` to handle this difference.

Reply:
@jeromelaban I don't see how this could leak. The Popup and PopupPanel seem to have the same life-time. So instead of exhausting the pool (because the popup seems to never be returned), we just store a strong reference.

The code has changed much since the use of this weak reference was added, and I suspect that this is no longer needed.

Reply:
@jeromelaban I moved to using `UNO_HAS_UIELEMENT_IMPLICIT_PINNING`.

I also moved to a simple `WeakReference` since renting from the pool without returning the instance back doesn't feel a good idea.

Reply:
It actually is a preferable to use the renting part, since DependencyObjects implement `IWeakReferenceProvider`, which automatically returns a self-managed weak reference from the popup:

See:

https://github.com/unoplatform/uno/blob/fd3e16196fdad21282e9a82ffdde44c7d9aaada0/src/Uno.UI/DataBinding/WeakReferencePool.cs#L60-L65

We could remove `WeakReferencePool.ReturnWeakReference(this, _popup);` call, which does not return anything in that context, though.